File: server/nodejs/src/business/domain/payments/payment-details/payment-status-output.ts
import { APIError } from './api-error';

export type PaymentStatusOutput = {
    errors?: APIError[] | null;
    isAuthorized?: boolean | null;
    isCancellable?: boolean | null;
    isRefundable?: boolean | null;
    statusCategory?: string | null;
    statusCode?: number | null;
    statusCodeChangeDateTime?: string | null;
};

File: server/python/app/application/domain/payments/payment_details/payment_status_output.py
from dataclasses import dataclass
from typing import Optional, List

from app.application.domain.payments.payment_details.api_error import APIError

@dataclass
class PaymentStatusOutput:
    errors: Optional[List[APIError]] = None
    is_authorized: Optional[bool] = None
    is_cancellable: Optional[bool] = None
    is_refundable: Optional[bool] = None
    status_category: Optional[str] = None
    status_code: Optional[int] = None
    status_code_change_date_time: Optional[str] = None